tools/c7n_azure/tests/test_tags.py

```diff
@@ -43,28 +43,29 @@ def setUp(self):
 
     def tearDown(self):
         self.after_tags = self.get_tags(self.rg_name, self.vm_name)
+        tags_to_remove = [k for k in self.after_tags.keys() if k not in self.before_tags.keys()]
+
+        if tags_to_remove:
+            if (self.after_tags != self.before_tags):
+                p = self.load_policy({
+                    'name': 'test-azure-tag',
+                    'resource': 'azure.vm',
+                    'filters': [
+                        {'type': 'value',
+                         'key': 'name',
+                         'op': 'eq',
+                         'value_type': 'normalize',
+                         'value': 'cctestvm'}
+                    ],
+                    'actions': [
```

Comment:
per Kapil's feedback in different PR, we should be using SDK client instead of the policy for the cleanup

Reply:
Agreed, I just can't rewrite every test.  In this case it is all the original code, I just moved the list comprehension from inside the action to above the policy because teardown was failing the test suite any time there were no tags to change.